codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -79,6 +80,27 @@ public void streamInactive(Http2Stream stream) {
             }
 
             @Override
+            public void streamHalfClosed(Http2Stream stream) {
+                if (State.HALF_CLOSED_LOCAL.equals(stream.state())) {
+                    /**
+                     * Any pending frames can never be written, clear and
```

Comment:
Ah I see, it's a copy and paste of the comment from `streamInactive` above.  Suggested way to clean this up...

1. make a private method called `streamClosedForWrite(stream)` and call it from both places.  It will just do `state(stream).clear()`.
2. Add a comment to `streamClosedForWrite` to say something like `Any pending frames for this stream can no longer be written, so clear the pending queue and call each frame's error handler`.